### ISSUE TITLE
BUG: np.cov with rowvar=False returns the wrong shape for N=1

### ISSUE THREAD

**Issue by jakevdp** [numpy/numpy#27658]
### Describe the issue:

For one observation of three variables, `np.cov` with `rowvar=True` returns the following:
```python
>>> x = np.ones((3, 1))
>>> np.cov(x, ddof=0, rowvar=True)
array([[0., 0., 0.],
       [0., 0., 0.],
       [0., 0., 0.]])
```
For the same computation laid-out with `rowvar=False`, the return value is a scalar:
```python
>>> np.cov(x.T, ddof=0, rowvar=False)
array(0.)
```
This should return a 3x3 matrix of zeros, same as the computation above.

The problem is the special-casing of `X.shape[0] == 1` in this line, which seems aimed at handling the `x.ndim == 1` case: https://github.com/numpy/numpy/blob/70fde29fdd4d8fcc6098df7ef8a34c84844e347f/numpy/lib/_function_base_impl.py#L2739-L2740

It looks like this behavior was introduced in 959f36c04ce8ca0b7bc44bb6438bddf162ad2db9, 19 years ago.

### Python and NumPy Versions:
```
$ python -c "import sys, numpy; print(numpy.__version__); print(sys.version)"
2.1.1
3.12.3 (v3.12.3:f6650f9ad7, Apr  9 2024, 08:18:47) [Clang 13.0.0 (clang-1300.0.29.30)]
```
### Runtime Environment:
```
$ python -c "import numpy; numpy.show_runtime()"
[{'numpy_version': '2.1.1',
  'python': '3.12.3 (v3.12.3:f6650f9ad7, Apr  9 2024, 08:18:47) [Clang 13.0.0 '
            '(clang-1300.0.29.30)]',
  'uname': uname_result(system='Darwin', node='jmdg-macbookpro.roam.internal', release='23.6.0', version='Darwin Kernel Version 23.6.0: Wed Jul 31 20:49:39 PDT 2024; root:xnu-10063.141.1.700.5~1/RELEASE_ARM64_T6000', machine='arm64')},
 {'simd_extensions': {'baseline': ['NEON', 'NEON_FP16', 'NEON_VFPV4', 'ASIMD'],
                      'found': ['ASIMDHP'],
                      'not_found': ['ASIMDFHM']}}]
```

**Comment by WarrenWeckesser** [comment]
FYI: This looks like a duplicate of https://github.com/numpy/numpy/issues/20951

**Comment by WarrenWeckesser** [comment]
> FYI: This looks like a duplicate of https://github.com/numpy/numpy/issues/20951

On closer inspection, I see that it is not the same, and #20951 is not fixed by #27661.
